### ISSUE TITLE
MathSAT / OSX: Hardcoded library path in 5.5.1

### ISSUE THREAD

**Issue by marcogario** [pysmt/pysmt#483]
Incorrect dylib object in _mathsat.so on OSX.

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "mathsat.py", line 21, in <module>
    _mathsat = swig_import_helper()
  File "mathsat.py", line 20, in swig_import_helper
    return importlib.import_module('_mathsat')
  File "/usr/local/Cellar/python@2/2.7.15/Frameworks/Python.framework/Versions/2.7/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
ImportError: dlopen(./_mathsat.so, 2): Library not loaded: /Users/griggio/Documents/src/mathsat_release/build/libmathsat.dylib
  Referenced from: /Users/travis/python_bindings/msat/_mathsat.so
  Reason: image not found
```

We either need to change the installer to rewrite this (we already had this issue with CVC4 on mac, if I recall correctly), or this needs to be fixed upstream.

**Comment by marcogario** [comment]
@agriggio Can you take a look?

This is how I plan to solve this: https://stackoverflow.com/questions/40325858/how-to-change-rpath-on-os-x

**Comment by marcogario** [comment]
This is solved in our installer in PR #487 . However, this should be fixed upstream.
